www-servers/mongrel2: Fix build failure with mbedtls-2.3

jasom opened the pull request:
Patch header-file include order to work with newer mbedtls versions

Gentoo-Bug: https://bugs.gentoo.org/590512

Package-Manager: portage-2.2.26

monsieurp commented:
@gentoo/proxy-maint

monsieurp commented on www-servers/mongrel2/mongrel2-1.11.0-r1.ebuild:
```diff
@@ -0,0 +1,31 @@
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
```
Might as well bump to EAPI 6 whilst at it. What do you think?

jasom commented:
EAPI version is bumped now.

mgorny commented on www-servers/mongrel2/mongrel2-1.11.0-r1.ebuild:
```diff
@@ -0,0 +1,31 @@
+
+src_prepare() {
+	cp "${FILESDIR}/systemtls.mak" Makefile || die
+	epatch "${FILESDIR}/${P}-polarssl-platform-590512.patch"
```
in EAPI 6, you can use eapply instead of epatch and therefore get rid of eutils.

jasom commented:
I switched to using eapply and squashed the two commits together (since now the patching and upgrade to EAPI 6 are logically part of the same change).

jasom commented:
Anything else I need do for this?

jasom commented:
Landed as 1a12b8300619c92eabcb63ab15e24422ce6ef892
